pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -154,6 +166,7 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 				return requeue, err
 			}
 		}
+		delete(hyd.Annotations, constant.AnnoHostingVersion)
```

Comment:
Here we remove the annotation. This flow may run during every reconcile, but the annotation is not written to the object..